google-cloud-dataproc/synth.py

```diff
@@ -38,11 +38,40 @@
 s.copy(v1_library / '.yardopts')
 s.copy(v1_library / 'google-cloud-dataproc.gemspec', merge=ruby.merge_gemspec)
 
-# https://github.com/googleapis/gapic-generator/issues/2232
-s.replace(
-    'lib/google/cloud/dataproc/v1/cluster_controller_client.rb',
-    '\n\n(\\s+)class OperationsClient < Google::Longrunning::OperationsClient',
-    '\n\n\\1# @private\n\\1class OperationsClient < Google::Longrunning::OperationsClient')
+v1beta2 = gapic.ruby_library(
+    'dataproc', 'v1beta2', config_path='/google/cloud/dataproc/artman_dataproc_v1beta2.yaml',
+    artman_output_name='google-cloud-ruby/google-cloud-dataproc'
+)
+s.copy(v1beta2 / 'lib/google/cloud/dataproc/v1beta2')
+s.copy(v1beta2 / 'lib/google/cloud/dataproc/v1beta2.rb')
+s.copy(v1beta2 / 'acceptance/google/cloud/dataproc/v1beta2')
+s.copy(v1beta2 / 'test/google/cloud/dataproc/v1beta2')
+
+for version in ['v1', 'v1beta2']:
+    # https://github.com/googleapis/gapic-generator/issues/2232
+    s.replace(
+        f'lib/google/cloud/dataproc/{version}/cluster_controller_client.rb',
+        '\n\n(\\s+)class OperationsClient < Google::Longrunning::OperationsClient',
+        '\n\n\\1# @private\n\\1class OperationsClient < Google::Longrunning::OperationsClient')
+
+    # https://github.com/googleapis/gapic-generator/issues/2492
+    s.replace(
+        [
+            'lib/google/cloud/dataproc.rb',
```

Comment:
This will run the replacement twice in this file. Which should be harmless in this case, but I'd rather it be cleaner, just to avoid future issues.

In other synth files where we generate multiple versions, we generally just go through once (rather than nesting in a loop), and for each replacement, expand the list of files to include all the versions. In some cases, you can use glob syntax (e.g. `lib/google/cloud/dataproc/v*/workflow_template_service_client.rb`)